### PR TITLE
bpo-31752: Fix possible crash in timedelta constructor called with custom integers.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Bad remainder in divmod() in intermediate calculations caused an assertion failure.

https://bugs.python.org/issue31752

### CODE REVIEW

File: Lib/test/datetimetester.py [hunk 1]
```diff
@@ -886,6 +886,26 @@ def as_integer_ratio(self):
...
                 timedelta() * get_bad_float(bad_ratio)
 
+    def test_issue31752(self):
+        # The interpreter shouldn't crash because divmod() returns negative
+        # remainder.
+        class BadInt(int):
+            def __mul__(self, other):
+                return Prod()
+
+        class Prod:
```
Comment (haypo) [inline Lib/test/datetimetester.py]: Maybe inherit from int?

Comment (vstinner) [inline Lib/test/datetimetester.py:896]: Maybe inherit from int as well?

**Review by vstinner — APPROVED** [review]
LGTM, except of a minor comment.

File: Lib/test/datetimetester.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/datetimetester.py:896]: No, it shouldn't be inherited from int, otherwise `__radd__` would not be called.

Comment (vstinner) [reply Lib/test/datetimetester.py:896]: Oh ok.

**Review by terryjreedy — DISMISSED** [review]
On Win10, this changes a noisy crash, with crash box and printed message, to a silent crash
f:\dev\3x>python f:/python/mypy/tem.py
Running Debug|Win32 interpreter...

f:\dev\3x>  # printed immediately after above

 which to me is worse.  I verified that datetimetester.py has the new test_issue31752.

**Comment by serhiy-storchaka** [comment]
What is `f:/python/mypy/tem.py`? I just have installed mypy, but don't see the file `tem.py` there. Tests are passed on Windows to me.

**Comment by terryjreedy** [comment]
`mypy` is my personal directory of miscellaneous .py files, which a have had for at least a decade.  I guess I should rename it.  tem.py should have been the code in the initial post with a print added.  I should have been clearer.  

*Without* the patch, with a fresh build,
 `python -m test test_datetime` passes (and ditto for whole suite, without `test_datetime` added)
`python -m unittest test.test_datetime` passes
`python -m test.test_datetime` crashes in test_strftime with 'Debug assertion failed' in line 1184 of `min\kernal\crts\ucrt\src\appcrt\time\wcsftime.cpp`.  I don't know what to make of this. 

I will apply the patch again, re-build, and try to reproduce what I think I did before, keep better notes, and report again.

**Comment by terryjreedy** [comment]
This time, printing the time delta of 1 in your opening code works instead of crashing.
`python -m test.test_datetime` still crashes, but that must be a different issue.

**Pull request merged**